conduit/plugins/processors/filterprocessor/expression/expression.go

```diff
@@ -101,18 +114,21 @@ func makeUnsignedExpression(searchStr string, expressionType FilterType) (Expres
 }
 
 // MakeExpression creates an expression based on an expression type
-func MakeExpression(filterType FilterType, expressionSearchStr string, target interface{}) (exp Expression, err error) {
+func MakeExpression(expressionType Type, expressionSearchStr string, target interface{}) (exp Expression, err error) {
+	if _, ok := TypeMap[expressionType]; !ok {
+		return nil, fmt.Errorf("expression type (%s) is not supported", expressionType)
+	}
+
 	switch t := target.(type) {
 	case uint64:
-		return makeUnsignedExpression(expressionSearchStr, filterType)
+		return makeUnsignedExpression(expressionSearchStr, expressionType)
 	case int64:
-		return makeSignedExpression(expressionSearchStr, filterType)
+		return makeSignedExpression(expressionSearchStr, expressionType)
 	case string:
-		if filterType == EqualToFilter {
-			// Equal to for strings is a special case of the regex pattern.
-			expressionSearchStr = fmt.Sprintf("^%s$", regexp.QuoteMeta(expressionSearchStr))
+		if expressionType == EqualTo {
+			return &stringEqualExpression{Str: expressionSearchStr}, nil
 		}
```

Comment:
Switching from a special regex expression to string equality was an 8-12% performance increase according to the benchmark.